src/python/foglamp/device/exceptions.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Device service exceptions module """
+
+__author__ = "Stefano Simonelli"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+class InvalidServiceNameError(Exception):
```

Comment:
let's make a  def DeviceServiceError or DeviceServiceException with init having code and message? and subclass each next with that?